### PR TITLE
Python formatting

### PR DESCRIPTION
**Pull request by travisdowns**
This series enforces python formatting via a CI check, starting with addr2line files.

It also formats those two files. The formatter is black which seems like the dominant formatter these days and it adheres to (a subset of) PEP8 and seems generally in line with our own formatting. The two config changes I made from the defaults were to extend the line length to 120 (from 88) and to disable "string conversion" (which changes almost all `'` to `"`). This reduces the churn a lot but I'm also happy to remove those (or just the latter) if we want to go totally default.

### CODE REVIEW

**Comment by travisdowns** [comment]
You can see an example of the CI run in this change, and [here's an example](https://github.com/travisdowns/seastar/actions/runs/10914503050/job/30292722957) of what a failing run looks like, showing the formatting delta.

I've tested that using black extension in vscode results in the same format (in fact, that's how I formatted the two files here).

The CI run is fast, taking < 10 seconds to complete.

**Review by tchaikov — APPROVED** [review]
lgtm.

File: .github/workflows/python-lint.yaml
Comment (travisdowns) [reply .github/workflows/python-lint.yaml:11]: They do, but don't provide a justification. 

I have a strong bias towards explicit versions, especially in a scenario like this where there is not really any motivation to stay up to date. Tags that change like stable mean from one run to another the CI behavior can change and an arbitrary change will be selected as the victim in that case, even if the changes are unrelated. 

That said, as an external contributor I'm not going to die on this hill so if you prefer stable, happy to make that change.

Comment (tchaikov) [reply .github/workflows/python-lint.yaml:11]: i see your point. and it makes sense to me. let's wait for maintainers' reviews.

File: scripts/pyproject.toml
Comment (xemul) [inline scripts/pyproject.toml:3]: I'm not an expert, but I didn't find where this file is used by next patching

Comment (tchaikov) [reply scripts/pyproject.toml:3]: this file is read by the github action of `psf/black`. see https://github.com/psf/black/blob/bbfdba3a5e4f0b3105ec9526c51e769866b28f1f/action/main.py#L59C20-L59C34

File: .github/workflows/python-lint.yaml
Comment (avikivity) [reply .github/workflows/python-lint.yaml:11]: Agree with @travisdowns

**Pull request closed without merging**
